vscode/oso/client/src/telemetry.ts

```diff
@@ -0,0 +1,135 @@
+import { createHash } from 'crypto';
+
+import { ExtensionContext, env, Uri, OutputChannel } from 'vscode';
+import * as Mixpanel from 'mixpanel';
+import {
+  Diagnostic,
+  DiagnosticSeverity as Severity,
+} from 'vscode-languageclient';
+
+export const telemetryEventsKey = 'events';
+// Flush telemetry events in batches every hour.
+export const TELEMETRY_INTERVAL = 1000 * 60 * 60;
+
+const hash = (contents: string) =>
+  createHash('sha256').update(contents).digest('base64');
```

Comment:
potentially overparanoid security engineer speaking here: 
we might consider prefixing the `contents` here with a domain-specific prefix `oso-telemetry:$CONTENTS` to prevent someone from leveraging existing rainbow table SHA256 datasets to work backwards from a hashed identifier.

Reply:
65ce867